### PR TITLE
Add warning about query performance

### PR DESCRIPTION
**Pull request by nefta-kanilmaz-by**

### CODE REVIEW

File: docs/spec/efficient_querying.rst [hunk 1]
```diff
@@ -54,6 +54,9 @@ Through restrictions on partitioned columns and indexed columns, Kartothek can
 significantly reduce the number of parquet files to look at. The selection of
 columns allows it to transfer only a subset of the data.
 
+.. warning::
+   When using the `in` operation to filter your data, avoid passing a long list
```
Comment (lr4d) [inline docs/spec/efficient_querying.rst:58]:
```suggestion
   When using the `in` operation to filter your data, avoid passing a long (i.e. >100 elements) list
```

Comment (nefta-kanilmaz-by) [reply docs/spec/efficient_querying.rst:58]: Done

**Comment by fjetter** [comment]
I'm wondering if this is still relevant with https://github.com/JDASoftwareGroup/kartothek/pull/358

**Review by lr4d — APPROVED** [review]

**Comment by fjetter** [comment]
I think this is still relevant since there may be other perf related issues, e.g. when using predicate pushdown. Thanks @NeroCorleone

**Pull request merged**